cli/internal/reference/compref/compref.go

```diff
@@ -185,25 +185,70 @@ func Parse(input string) (*Ref, error) {
 		return nil, fmt.Errorf("invalid component name %q in %q, must match %q", ref.Component, originalInput, ComponentRegex)
 	}
 
-	// Step 6: Resolve type if not explicitly given
+	// Step 6: Build repository object using ParseRepository
+	var repositorySpec string
+	if ref.Type != "" {
+		repositorySpec = ref.Type + "::" + input
+	} else {
+		repositorySpec = input
+	}
+
+	repository, err := ParseRepository(repositorySpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse repository: %w", err)
+	}
+
+	ref.Repository = repository
+
+	// Extract the type from the parsed repository for consistency
 	if ref.Type == "" {
```

Comment:
im wondering if you shouldn't default this type field always. is there a reason is should be only set when not specified?

Reply:
I think for now this is fine. resolving as it was a nit anyhow